### ISSUE TITLE
Custom JSON logger and duplicate log messages

### ISSUE THREAD

**Issue by wangjinbei** [kdeldycke/click-extra#989]
Hi, when I use https://kdeldycke.github.io/click-extra/logging.html#custom-logger

My terminal outputs two logs, is there a problem where I am processing?
My intention: to format all log content in json format so that I can analyze and notify using the log collector .

Also what can I do to define the `Load configuration matching foo.ini` in json format as well?

demo.py:

```python
import logging

from click import pass_context
from click_extra import extra_command, extra_basic_config, verbosity_option

extra_basic_config(
    logger_name="app_logger",
    format='{{"time": "{asctime}", "name": "{name}", "level": "{levelname}", "msg": "{message}"}}',
    force=True,
    level=logging.INFO,
)


@extra_command
@pass_context
@verbosity_option(default_logger="app_logger",)
def cli(config):
    logging.getLogger("app_logger").info("This is an info message from the root logger.")
```

output:
```shell-session
Load configuration matching foo.ini
{"time": "2024-06-18 10:35:50,018", "name": "app_logger", "level": "info", "msg": "This is an info message from the root logger."}
info: This is an info message from the root logger.
```

**Comment by kdeldycke** [comment]
I can reproduce your case with the latest Click Extra (v4.11.7), so nothing's wrong on your side. It seems that the duplicate log has always been there since I introduced unittests with the documentation in v4.3.0: https://github.com/kdeldycke/click-extra/commit/ec4243cc2422520ec4603e560f0abcc9c76d7ea0#diff-1a5e57d66b08bec5a71860d67f06912a004f03368c36910099d4ffae12970c7dR230-R231

And as you can see it is how that's being documented at https://kdeldycke.github.io/click-extra/logging.html#custom-logger

As for the `Load configuration matching` message, it seems there is a chicken and egg problem involved, as per the comments in the source code: https://github.com/kdeldycke/click-extra/blob/e495e3b0e19b8ebf46086e6165f28ce8a1f70885/click_extra/config.py#L394-L397

Still, your use-case of producing JSON for all logs is interesting. Let me dive into that. Maybe it's just a matter of setting up the right configuration.

**Comment by kdeldycke** [comment]
The issue regarding the duplicate message has been documented at: https://github.com/kdeldycke/click-extra/blob/5ea95cfb9e70dc60d50fbe482b2319f4f467fbc3/docs/logging.md?plain=1#L338-L375

Tl;Dr: you need to `logging.getLogger("app_logger").propagate = False`.
